### PR TITLE
Have findProductList() Make Async Calls in Parallel.

### PR DESCRIPTION
**Pull request by danielcliu-ifixit**
## Overview
Before, we had a call to strapi and a conditional call to iFixit API afterwards. This makes it so the two requests are done at the same time so we can cut out the overhead of the faster request. This should make the two calls take less time overall as one is not conditional on the other and they are practically always both called anyway.

## CR
Pretty simple

## QA
Do product lists still load? Are they any faster? I've found a slight improvement when running locally (~60 ms speed up) but couldn't find that same performance gain in lighthouse for whatever reason.

Closes: #525

### CODE REVIEW

File: frontend/lib/ifixit-api/devices.ts [hunk 1]
```diff
@@ -7,6 +7,9 @@ export async function fetchDeviceWiki(
 ): Promise<DeviceWiki | null> {
    const deviceHandle = getDeviceHandle(deviceTitle);
    try {
+      if(!deviceHandle) {
+         throw new Error(`deviceHandle cannot be a blank string!`);
+      }
```
Comment (dhmacs) [inline frontend/lib/ifixit-api/devices.ts:12]:
```suggestion
     invariant(deviceHandle.length > 0, "deviceHandle cannot be a blank string");
```

Comment (masonmcelvain) [reply frontend/lib/ifixit-api/devices.ts:12]: Have you seen cases where this is called with a blank string @danielcliu-ifixit?

File: frontend/models/product-list/index.ts [hunk 2]
```diff
@@ -47,16 +47,17 @@ export type {
 export async function findProductList(
    filters: ProductListFiltersInput
 ): Promise<ProductList | null> {
-   const result = await strapi.getProductList({
-      filters,
-   });
+   const deviceTitle = filters.deviceTitle?.eq ?? '';
```
Comment (masonmcelvain) [inline frontend/models/product-list/index.ts:50]: This seems like a change in behavior. We never want to use `productList.deviceTitle` anymore?

File: frontend/models/product-list/index.ts [hunk 3]
```diff
@@ -47,16 +47,17 @@ export type {
...
 
+   const [result, deviceWiki] =
+      await Promise.all([
+         strapi.getProductList({
+            filters,
+         }),
+         fetchDeviceWiki(deviceTitle)
+      ]);
```
Comment (masonmcelvain) [inline frontend/models/product-list/index.ts:58]: I like how succinct this is. I would personally avoid calling `fetchDeviceWiki(deviceTitle)` if `deviceTitle === ''`, but the result is the same so I'm fine with it either way.

```suggestion
   const [result, deviceWiki] =
      await Promise.all([
         strapi.getProductList({
            filters,
         }),
         deviceTitle ? fetchDeviceWiki(deviceTitle) : null,
      ]);
```

**Review by masonmcelvain — CHANGES_REQUESTED** [review]
CR ✌🏻 with a question about `productList.deviceTitle`

**Comment by k0rvusk0r4x** [comment]
- I'm seeing an improvement in speed-based metrics (time-to xxx) and in the Speed Index specifically. 
- The page load is working without issue.

QA 😺

File: frontend/models/product-list/index.ts (hunk 2 above)
Comment (danielcliu-ifixit) [reply frontend/models/product-list/index.ts:50]: Apparently the strapi api is case-sensitive so I'm not worried about the two being different. I don't think they'll be a behavior change

File: frontend/models/product-list/index.ts (hunk 3 above)
Comment (danielcliu-ifixit) [reply frontend/models/product-list/index.ts:58]: I like handling it in the function so if anyone uses the function in future they don't have to account for their input being `== null`

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻

**Comment by deltuh-vee** [comment]
Carry Over QA 🎬

**Pull request merged**